### PR TITLE
feat: Create a BaseButtonComponent to handle dynamic button UI and functionality (FM-289)

### PR DESCRIPTION
**Pull request by woodsjohn-codev**
# Changes
- Created a BaseButtonComponent to handle buttons

# How to test
- import BaseButtonComponent to any components
- for example: in pause button:
```
import {PAUSE_BTN_IMG} from '@constants';
import gameStateService from '@gameStateService';
import {BaseButtonComponent} from './base-button/base-button-component';

export default class PauseButton extends BaseButtonComponent {
  constructor() {
    super({
      id: 'pause-button',
      className: 'pause-button-image',
      onClick: () => {
        console.log('Pause button clicked or touched');
        gameStateService.publish(
          gameStateService.EVENTS.GAME_PAUSE_STATUS_EVENT,
          true, // isGamePaused
        );
      },
      imageSrc: PAUSE_BTN_IMG,
      imageAlt: 'Pause Icon',
    });
  }
}
```
- then call pause button component for example in gameplay-scene.ts
```
new PauseButton();
```
Ref: [FM-289](https://curiouslearning.atlassian.net/browse/FM-289)

[FM-289]: https://curiouslearning.atlassian.net/browse/FM-289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/buttons/base-button/base-button-component.ts [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+// import {PAUSE_BTN_IMG} from '@constants';
+// import gameStateService from '@gameStateService';
+
+// import {BaseButtonComponent} from './base-button/base-button-component';
+
+// export default class PauseButton extends BaseButtonComponent {
+//   constructor() {
+//     super({
+//       id: 'pause-button',
+//       className: 'pause-button-image',
```
Comment (ashwinnair-chimple) [inline src/components/buttons/base-button/base-button-component.ts:73]: Remove example usage from base button component and implement the pause button functionality on gameplay screen instead.

File: src/components/buttons/base-button/base-button-component.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
+interface ButtonOptions {
```
Comment (miguelccodev) [inline src/components/buttons/base-button/base-button-component.ts:1]: Add `export` to interface.

File: src/components/buttons/base-button/base-button-component.ts (hunk 1 above)
Comment (miguelccodev) [reply src/components/buttons/base-button/base-button-component.ts:73]: @woodsjohn-codev this is out of scope. You can provide a simple example instead. But this is basically another task already.

**Review by miguelccodev — CHANGES_REQUESTED** [review]

File: src/components/buttons/base-button/base-button-component.ts (hunk 1 above)
Comment (woodsjohn-codev) [reply src/components/buttons/base-button/base-button-component.ts]: please check again @miguelccodev

File: src/components/buttons/base-button/base-button-component.ts (hunk 2 above)
Comment (woodsjohn-codev) [reply src/components/buttons/base-button/base-button-component.ts]: done. please check again @miguelccodev

**Review by ashwinnair-chimple — APPROVED** [review]
Approved

**Pull request merged**